Application/EdFi.Ods.Features/XsdMetadata/XsdMetadataController.cs

```diff
@@ -73,36 +78,57 @@ public IActionResult Get()
 
         [HttpGet]
         [Route("{schema}/files")]
-        public IActionResult GetFiles([FromRoute] string schema)
+        public IActionResult GetFiles([FromRoute] XsdMetadataRequest request)
         {
             if (!_isEnabled)
             {
                 return NotFound();
             }
 
-            var schemaInformation = _xsdFileInformationProvider.XsdFileInformationByUriSegment(schema);
+            var schemaInformation = _xsdFileInformationProvider.XsdFileInformationByUriSegment(request.Schema);
 
             if (schemaInformation == default)
             {
                 return Ok(new Uri[0]);
             }
 
             var results = schemaInformation.SchemaFiles
-                .Select(
-                    x => new Uri(
-                        $"{_urlHelper.ActionLink("Get", ControllerContext.ActionDescriptor.ControllerName)}/{schemaInformation.SchemaNameMap.UriSegment}/{x}"))
+                .Select(x=> GetMetadataAbsoluteUrl(x, schemaInformation.SchemaNameMap.UriSegment))
                 .ToList();
-
+            
             if (!schemaInformation.IsCore())
             {
                 var coreInformation = _xsdFileInformationProvider.CoreXsdFileInformation();
 
-                results.AddRange(
-                    coreInformation.SchemaFiles
-                        .Select(
-                            x => new Uri(
-                                $"{_urlHelper.ActionLink("Get", ControllerContext.ActionDescriptor.ControllerName)}/{coreInformation.SchemaNameMap.UriSegment}/{x}"))
-                );
+                results.AddRange(coreInformation.SchemaFiles
+                        .Select(x => GetMetadataAbsoluteUrl(x, coreInformation.SchemaNameMap.UriSegment))
+                        .ToList());
```

Comment:
why adding the list you are forcing an extra enumeration.

Reply:
I removed it . Thanks